src/components/AppConfig/AppConfig.tsx

```diff
@@ -202,6 +203,25 @@ export const updatePlugin = (pluginId: string, data: Partial<PluginMeta>) => {
   return lastValueFrom(response);
 };
 
+export const savePluginSettings = (data: Partial<PluginMeta>) => {
+  console.log('Saving plugin settings', data)
+  const response = getBackendSrv().fetch({
+    url: `api/plugins/grafana-llm-app/resources/save-plugin-settings`,
+    method: 'POST',
+    data
+  });
+
+  return lastValueFrom(response);
+};
+
+export const updateAndSavePluginSettings = (pluginId: string, data: Partial<PluginMeta>) => {
+  const gcomPluginData = {
+    jsonData: data.jsonData,
+    secureJsonData: data.secureJsonData,
+  };
+  return Promise.all([updatePlugin(pluginId, data), savePluginSettings(gcomPluginData)]);
```

Comment:
🤔 Pros/cons of doing two calls from the client vs using the backend to call both places? Probably doesn't matter and this is fine to start with?

Reply:
Unless I'm misunderstanding, the frontend won't have access to `secureJsonData`, it's only available in the backend (otherwise it wouldn't be very secure!)

Reply:
The secureJsonData from the frontend would be everything that is updated upon save.

Reply:
I think Ben's point is that the frontend shouldn't be even trying to write to secureJsonData. 

I noticed there's a `Secrets` type on line 25 already, that is supposedly populated from the initial settings. What makes them secure is that the backend does not pass the secrets to the frontend at all, only a "true" value - see here for a working-ish dev instance:
<img width="771" alt="image" src="https://github.com/grafana/grafana-llm-app/assets/632769/08adbeb5-cb7d-42f1-b3d6-875bca69073d">

I think you can avoid trying to push secureJsonData from the frontend.

Reply:
This has now been updated a bit but the flow is:
1. Update gcom by calling an endpoint in the plugin with all jsonData and any new secureJsonData. The endpoint needs to merge the new secureJsonData with the old in order to not lose data in gcom.
2. Update the plugin settings in Grafana (which will then be pulled from gcom on the next restart). This is the normal plugin update and doesn't need any merging as Grafana handles that for us.

I don't love that the state has to be sent to both of those places, but I don't see a way around it either. Anyone else have a better idea?